File: PYTHON/EASY_PY_LEETCODE/PROB_2099_Find_Subsequence_of_Length_K_With_the_Largest_Sum.py
from typing import *
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        indexed_nums = list(enumerate(nums))  
        indexed_nums.sort(key=lambda x: x[1], reverse=True)  
        top_k_elements = sorted(indexed_nums[:k], key=lambda x: x[0]) 
        return [x[1] for x in top_k_elements]

File: JAVASCRIPT/EASY_JS_LEETCODE/PROB_2099_Find_Subsequence_of_Length_K_With_the_Largest_Sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var maxSubsequence = function(nums, k) {
    return nums.map((value, index) => ({ value, index }))
               .sort((x,y)=>y.value-x.value)
               .slice(0,k)
               .sort((m,n)=>m.index-n.index)
               .map((p)=>p.value);
};